tests/hvm_cpu.rs

```diff
@@ -0,0 +1,384 @@
+use hvm_lang::term::{parser, DefNames, DefinitionBook, Term};
+use hvmc::{ast::*, run::NumericOp, *};
+use insta::{assert_debug_snapshot, assert_snapshot};
+use std::fs;
+
+fn load_file(file: &str) -> String {
+  let path = format!("{}/tests/programs/{}", env!("CARGO_MANIFEST_DIR"), file);
+  fs::read_to_string(path).unwrap()
+}
+
+// Parses code and generate Book from hvm-core syntax
+fn parse_core(code: &str) -> Book {
+  do_parse_book(code)
+}
+
+// Parses code and generate DefinitionBook from hvm-lang syntax
+fn parse_lang(code: &str) -> DefinitionBook {
+  parser::parse_definition_book(code).unwrap()
+}
+
+// Loads file and generate DefinitionBook from hvm-lang syntax
+fn load_lang(file: &str) -> DefinitionBook {
+  let code = load_file(file);
+  parse_lang(&code)
+}
+
+// For every pair in the map, replaces all matches of a string the other string
+fn replace_template(mut code: String, map: &[(&str, &str)]) -> String {
+  for (from, to) in map {
+    code = code.replace(from, to);
+  }
+  code
+}
+
+trait Normal {
```

Comment:
traits are supposed to encapsulate a lot of similar logic, the only similar logic now, is the name of the method and its argument. I think you can remove the trait, and implement the method(or function) without needing a trait and this inner Result type